### PR TITLE
IC-1133: Add referral details to SP referral view

### PR DESCRIPTION
**Pull request by Gweaton**
## What does this pull request do?

Adds all of the information we capture in the Refer journey (and a couple of hardcoded bits) to the SP referral screen.

We'll want to update the hardcoded data once we've managed to fetch it from Delius (as part of other tasks).

## What is the intent behind these changes?

To play back all the necessary data to SPs before they assign a referral.

## Screenshot (data a bit wonky due to various hardcoded mocks)
image

### CODE REVIEW

File: server/services/interventionsService.test.ts [hunk 1]
```diff
@@ -946,7 +946,7 @@ pactWith({ consumer: 'Interventions UI', provider: 'Interventions Service' }, pr
     },
     assignedTo: null,
     referenceNumber: 'HDJ2123F',
-    referral: {
+    referralFields: {
```
Comment (tomsmyers) [inline server/services/interventionsService.test.ts:949]: this is a breaking API change. do we really need to do that?

File: server/routes/serviceProviderReferrals/showReferralPresenter.ts [hunk 2]
```diff
@@ -17,30 +17,67 @@ export default class ShowReferralPresenter {
...
+      },
+    ]
+  }
+
   readonly serviceUserNotificationBannerArgs = {
     titleText: 'Service user details',
     html:
       `<p class="govuk-notification-banner__heading">${this.serviceUser.firstName} ${this.serviceUser.surname}<p>` +
-      `<p>Date of birth: ${this.serviceUserDateOfBirth}</p>` +
+      `<p>Date of birth: ${ShowReferralPresenter.govukFormattedDateFromStringOrNull(
```
Comment (tomsmyers) [inline server/routes/serviceProviderReferrals/showReferralPresenter.ts:116]: do we want to display something is this isn't present? like "dob unknown" or something?

Comment (tomsmyers) [reply server/routes/serviceProviderReferrals/showReferralPresenter.ts:116]: nvm, seen output below

**Review by tomsmyers — CHANGES_REQUESTED** [review]
looks good - just want to follow up on that API change

File: server/services/interventionsService.test.ts (hunk 1 above)
Comment (Gweaton) [reply server/services/interventionsService.test.ts]: Ok, I've removed this rename.

Comment (tomsmyers) [reply server/services/interventionsService.test.ts]: thanks!

**Review by tomsmyers — APPROVED** [review]

**Pull request merged**